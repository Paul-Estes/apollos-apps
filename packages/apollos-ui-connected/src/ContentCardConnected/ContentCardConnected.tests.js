import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import {
  IntrospectionFragmentMatcher,
  InMemoryCache,
} from 'apollo-cache-inmemory';
import introspectionQueryResultData from 'apolloschurchapp/src/client/fragmentTypes.json';

import { FeaturedCard } from '@apollosproject/ui-kit';

import { Providers } from '../utils/testUtils';
import GET_CONTENT_CARD from './getContentCard';

import ContentCardConnected from './ContentCardConnected';

const fragmentMatcher = new IntrospectionFragmentMatcher({
  introspectionQueryResultData,
});

describe('ContentCardConnected component', () => {
  it('renders ContentCardConnected', async () => {
    const mock = {
      request: {
        query: GET_CONTENT_CARD,
        variables: {
          contentId: 'DevotionalContentItem:d395278cd4b68e074ca4e595c8feab6d',
        },
      },
      result: {
        data: {
          node: {
            id: 'DevotionalContentItem:d395278cd4b68e074ca4e595c8feab6d',
            title: 'God sees who you can be not who you are',
            parentChannel: {
              id: 'ContentChannel:559b23fd0aa90e81b1c023e72e230fa1',
              name: 'Devotional',
              iconName: 'text',
              __typename: 'ContentChannel',
            },
            hyphenatedTitle: 'Fake-hyphenated-title',
            node: {
              id: 'DevotionalContentItem:d395278cd4b68e074ca4e595c8feab6d',
              coverImage: {
                name: 'Square Image',
                sources: [
                  {
                    uri:
                      'https://res.cloudinary.com/apollos/image/fetch/c_limit,f_auto,w_1600/https://apollosrock.newspring.cc/GetImage.ashx%3Fguid%3D31af1a61-360c-4b1e-8e62-45517c06a9a2',
                    __typename: 'ImageMediaSource',
                  },
                ],
                __typename: 'ImageMedia',
              },
              parentChannel: {
                id: 'ContentChannel:559b23fd0aa90e81b1c023e72e230fa1',
                name: 'Devotional',
                iconName: 'text',
                __typename: 'ContentChannel',
              },
              title: 'God sees who you can be not who you are',
              sharing: {
                url:
                  'https://apollosrock.newspring.cc/devotional/god-sees-who-you-can-be-not-who-you-are',
                message:
                  'God sees who you can be not who you are - Life is challenging enough.',
                title: 'Share via ...',
                __typename: 'SharableContentItem',
              },
              __typename: 'DevotionalContentItem',
            },
            summary: 'bla bla bla',
            isLiked: false,
            theme: null,
            coverImage: {
              sources: [
                {
                  uri:
                    'https://apollosrock.newspring.cc/GetImage.ashx?guid=993efb47-47ec-4dec-a3d3-47bcacfbd58e',
                  __typename: 'ImageMediaSource',
                },
              ],
              __typename: 'ImageMedia',
            },
            __typename: 'DevotionalContentItem',
          },
        },
      },
    };
    const tree = renderer.create(
      <Providers
        mocks={[mock]}
        cache={
          new InMemoryCache({
            fragmentMatcher,
          })
        }
      >
        <ContentCardConnected
          Component={FeaturedCard}
          contentId={'DevotionalContentItem:d395278cd4b68e074ca4e595c8feab6d'}
          isLoading={false}
        />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
