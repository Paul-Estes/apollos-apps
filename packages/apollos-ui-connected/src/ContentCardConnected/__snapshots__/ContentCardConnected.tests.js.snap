// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCardConnected component renders ContentCardConnected 1`] = `
<View
  cardColor="#00676D"
  isLoading={false}
  style={
    Object {
      "backgroundColor": "#00676D",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(165, 165, 165, 0.6)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#A5A5A5",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#A5A5A5",
            "width": "100%",
          }
        }
      />
      <BVLinearGradient
        colors={
          Array [
            1711302509,
            4278216557,
          ]
        }
        endPoint={
          Object {
            "x": 0,
            "y": 1,
          }
        }
        locations={
          Array [
            0,
            0.8,
          ]
        }
        overlayColor="#00676D"
        overlayType="featured"
        startPoint={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "marginTop": "-40%",
          "paddingBottom": 32,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <View
        hasSummary={false}
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "paddingTop": 16,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "marginRight": 16,
            }
          }
        >
          <Text
            numberOfLines={4}
            style={
              Object {
                "color": "#F8F7F4",
                "fontFamily": "InterUI-Black",
                "fontSize": 36,
                "lineHeight": 41.4,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
